core/src/main/java/org/elasticsearch/index/get/GetResult.java

```diff
@@ -307,10 +307,16 @@ public static GetResult fromXContentEmbedded(XContentParser parser) throws IOExc
                         fields.put(getField.getName(), getField);
                     }
                 } else {
-                    throwUnknownField(currentFieldName, parser.getTokenLocation());
+                    parser.skipChildren(); // skip potential inner objects for forward compatibility
                 }
+            } else if (token == XContentParser.Token.START_ARRAY) {
+                parser.skipChildren(); // skip potential inner arrays for forward compatibility
             }
         }
+
+        if (found == null) {
```

Comment:
which makes me wonder...shouldn't we do the same in the update api with 404? I'm afraid so... cause if the doc is missing we do return 404?